libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -1217,6 +1225,13 @@ bool AP_Arming::disarm(const AP_Arming::Method method)
     }
 #endif
 
+    AC_Fence *fence = AP::fence();
+    if (fence != nullptr) {
+        if(fence->auto_enabled() == AC_Fence::AutoEnable::ONLY_WHEN_ARMED) {
```

Comment:
I'm not sure this follows - we may have enabled the fence with a switch, for example....

However... existing behaviour...

Reply:
Agree, existing behaviour. Intention was to push the disarming from the higher layers of plane to a common area for all vehicles. 

It's certainly debatable, I would say if you have the enabled fence via switch then you wouldn't want to turn them off automatically, unless your `AUTOENABLE` param was explicitly set to do so.